common/db-scripts/db-DataGeneration/inventory_creation.sql

```diff
@@ -0,0 +1,382 @@
+        WHERE LOWER([tenant]_mod_inventory_storage.f_unaccent(jsonb ->> 'hrid'::text)) = LOWER(instance_hrid)
+      ) INTO conflict_found;
+      EXIT WHEN NOT conflict_found;
+    END LOOP;
+
+    LOOP
+      holding_hrid := 'cenho' || LPAD((FLOOR(RANDOM() * 10000000000))::TEXT, 11, '0');
+      SELECT EXISTS (
+        SELECT 1 FROM [tenant]_mod_inventory_storage.holdings_record 
+        WHERE LOWER([tenant]_mod_inventory_storage.f_unaccent(jsonb ->> 'hrid'::text)) = LOWER(holding_hrid)
+      ) INTO conflict_found;
+      EXIT WHEN NOT conflict_found;
+    END LOOP;
+
+    LOOP
+      item_hrid := 'cenit' || LPAD((FLOOR(RANDOM() * 10000000000))::TEXT, 11, '0');
+      SELECT EXISTS (
+        SELECT 1 FROM [tenant]_mod_inventory_storage.item 
+        WHERE LOWER([tenant]_mod_inventory_storage.f_unaccent(jsonb ->> 'hrid'::text)) = LOWER(item_hrid)
+      ) INTO conflict_found;
+      EXIT WHEN NOT conflict_found;
+    END LOOP;
+
+    -- Generate random barcode
+   LOOP
+      item_barcode := LPAD((FLOOR(RANDOM() * 1000000000000))::TEXT, 12, '0');
+      SELECT EXISTS (
+        SELECT 1 FROM [tenant]_mod_inventory_storage.item 
+        WHERE LOWER(jsonb ->> 'barcode'::text) = LOWER(item_barcode)
+      ) INTO conflict_found;
+      EXIT WHEN NOT conflict_found;
+    END LOOP;
+    
+    -- Create unique titles
+    instance_title := 'PerfTesting Instance ' || i;
+    
+    -- Get current timestamp
+    creation_date := NOW();
+    complete_updated_date := NOW();
+    
+    -- Insert into instance table
+    INSERT INTO [tenant]_mod_inventory_storage.instance (
+      id, jsonb, creation_date, created_by, instancestatusid, modeofissuanceid, instancetypeid, complete_updated_date
+    ) VALUES (
+      instance_id,
+      jsonb_build_object(
+        'id', instance_id,
+        'hrid', instance_hrid,
+        'tags', jsonb_build_object('tagList', jsonb_build_array()),
+        'notes', jsonb_build_array(),
+        'title', instance_title,
+        'series', jsonb_build_array(),
+        'source', 'FOLIO',
+        '_version', 1,
+        'editions', jsonb_build_array(),
+        'metadata', jsonb_build_object(
+          'createdDate', to_char(creation_date, 'YYYY-MM-DD"T"HH24:MI:SS.MS"Z"'),
+          'updatedDate', to_char(creation_date, 'YYYY-MM-DD"T"HH24:MI:SS.MS"Z"'),
+          'createdByUserId', created_by_user,
+          'updatedByUserId', created_by_user
+        ),
+        'subjects', jsonb_build_array(),
+        'languages', jsonb_build_array(),
+        'indexTitle', '123',
+        'identifiers', jsonb_build_array(),
+        'publication', jsonb_build_array(),
+        'contributors', jsonb_build_array(),
+        'catalogedDate', to_char(creation_date, 'YYYY-MM-DD'),
+        'staffSuppress', false,
+        'instanceTypeId', instance_type_id,
+        'previouslyHeld', false,
+        'classifications', jsonb_build_array(),
+        'instanceFormats', jsonb_build_array(),
```

Comment:
there are many empty arrays created by calling jsonb_build_array(). Since this is the case, could you create a variable, called "emtpyArray", set it once to json_build_array(), and use it as the value for any field that requires an empty array? That way it wouldn't need to call jsonb_build_array() over and over again and will make it faster.